### ISSUE TITLE
Column type doesn't have type information?

### ISSUE THREAD

**Issue by hlung** [groue/GRDB.swift#309]
> ℹ We use Github issues for bug reports, feature requests, general support, and questions about the library.
>
> Please fill out this template when filing an issue, and remove this header before submitting.

## What did you do?

> Creating a table and write/read some data. Seeing this column definition in the example. Like this...
```
enum Columns {
        static let id = Column("id")
        static let title = Column("title")
        static let latitude = Column("latitude")
        static let longitude = Column("longitude")
    }
```

## What did you expect to happen?

> `Column` type seems to only have name. When creating table, I have to supply the type of the column again. I think `Column` type should have had type information.
```
try db.create(table: tableName) { t in
          t.column(Feed.Columns.id.name, .integer).primaryKey()
          t.column(Feed.Columns.content.name, .text)
        }
```

## Environment

**GRDB flavor(s):** GRDB
**GRDB version:** 2.8.0
**Installation method:** CocoaPods
**Xcode version:** Version 9.2 (9C40b)
**Swift version:** 4
**Platform(s) running GRDB:** iOS
**macOS version running Xcode:** 10.13.3 (17D102)

**Comment by groue** [comment]
Hello @hlung,

> When creating table, I have to supply the type of the column again. I think Column type should have had type information.
>
> ```swift
> try db.create(table: tableName) { t in
>   t.column(Feed.Columns.id.name, .integer).primaryKey()
>   t.column(Feed.Columns.content.name, .text)
> }
> ```

I understand why you want the `Feed` type to provide the type of the database columns that supports it.

But the `Column` type is designed for the runtime (requests), not the definition of the database schema. I suggest that you change the way you define your database schema.

---

The recommended way to define the database schema are [migrations](http://github.com/groue/GRDB.swift/blob/master/README.md#migrations) that are independent from record types, and use strings:

```swift
var migrator = DatabaseMigrator()

migrator.registerMigration("initialVersion") { db in
    try db.create(table: "feeds") { t in
      t.column("id", .integer).primaryKey()
      t.column("name", .text)
    }
    ...
}

// Migrations for future versions will be inserted here:
//
// migrator.registerMigration(...) { db in
//     ...
// }

// Migrate the database:
try migrator.migrate(dbQueue)
```

Migrations help your application evolve its application records and database schema *independently*.

The Feed type may change over time, with new properties, renamed properties, deleted properties, etc. A future version of your application may even replace the Feed type and the "feeds" table with a more suitable type and table.

When this happens, string-based migrations allow you to not rewrite the code that defines your database schema, and has been already carefully written and tested. You know the [if it ain't broke, don't fix it](https://en.wiktionary.org/wiki/if_it_ain%27t_broke,_don%27t_fix_it) proverb.

Instead, when the records of your application evolve, you add a new migration that brings your database schema up to date.

---

To sum up:

- Don't make your record types responsible for their database table, because they are unable to deal with schema evolution. Don't define any `Feed.createTable()` method.
- Instead, use [migrations](http://github.com/groue/GRDB.swift/blob/master/README.md#migrations), because they are specifically designed to deal with schema evolution.
